plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -979,3 +981,43 @@ def process_main(self, item=None):
         if pipe.returncode != 0:
             result_string = '%s\n::\n%s' % (stdout, stderr)
             raise PulpCodedException(message=result_string)
+
+
+class GenerateRepoviewStep(platform_steps.PluginStep):
+    """
+    Generate the static HTML files for a given repository using the repoview command
+    """
+    def __init__(self, content_dir):
+        """
+        Initialize the step for creating sqlite files
+
+        :param content_dir: The base directory of the repository.  This directory should contain
+                            the repodata directory
+        :type content_dir: str
+        """
+        super(GenerateRepoviewStep, self).__init__(constants.PUBLISH_GENERATE_REPOVIEW_STEP)
+        self.description = _('Generating HTML files')
+        self.content_dir = content_dir
+
+    def is_skipped(self):
+        """
+        Check the repo for the config option to generate the HTML files.
+        Skip generation if the config option is not specified.
+
+        :returns: Whether or not generating HTML files has been enabled for this repository
+        :rtype: bool
+        """
+        return not self.get_config().get('repoview', False)
+
+    def process_main(self, item=None):
+        """
+        Call out to repoview command line in order to process the files.
+        """
+        pipe = subprocess.Popen('repoview --title %(repo)s %(content_dir)s' %
+                                {'content_dir': self.content_dir, 'repo': self.get_repo().id},
+                                shell=True, stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+        stdout, stderr = pipe.communicate()
+        if pipe.returncode != 0:
+            result_string = '%s\n::\n%s' % (stdout, stderr)
+            raise PulpCodedException(message=result_string)
```

Comment:
You'll need to add a new error code to ``plugins/pulp_rpm/plugins/error_codes.py``, as the default error message for PulpCodedException appears to be "A general pulp exception occurred" and it doesn't print any strings passed to it. Not having repoview installed got me:
```
pulp.server.controllers.repository:ERROR: (26566-93472) Exception caught from plugin during publish for repo [zoo]
pulp.server.controllers.repository:ERROR: (26566-93472) Traceback (most recent call last):
pulp.server.controllers.repository:ERROR: (26566-93472)   File "/home/vagrant/devel/pulp/server/pulp/server/controllers/repository.py", line 1082, in _do_publish
pulp.server.controllers.repository:ERROR: (26566-93472)     publish_report = publish_repo(transfer_repo, conduit, call_config)
pulp.server.controllers.repository:ERROR: (26566-93472)   File "/home/vagrant/devel/pulp/server/pulp/server/async/tasks.py", line 658, in wrap_f
pulp.server.controllers.repository:ERROR: (26566-93472)     return f(*args, **kwargs)
pulp.server.controllers.repository:ERROR: (26566-93472)   File "/home/vagrant/devel/pulp_rpm/plugins/pulp_rpm/plugins/distributors/yum/distributor.py", line 174, in publish_repo
pulp.server.controllers.repository:ERROR: (26566-93472)     return self._publisher.process_lifecycle()
pulp.server.controllers.repository:ERROR: (26566-93472)   File "/home/vagrant/devel/pulp/server/pulp/plugins/util/publish_step.py", line 562, in process_lifecycle
pulp.server.controllers.repository:ERROR: (26566-93472)     super(PluginStep, self).process_lifecycle()
pulp.server.controllers.repository:ERROR: (26566-93472)   File "/home/vagrant/devel/pulp/server/pulp/plugins/util/publish_step.py", line 159, in process_lifecycle
pulp.server.controllers.repository:ERROR: (26566-93472)     step.process()
pulp.server.controllers.repository:ERROR: (26566-93472)   File "/home/vagrant/devel/pulp/server/pulp/plugins/util/publish_step.py", line 249, in process
pulp.server.controllers.repository:ERROR: (26566-93472)     self._process_block()
pulp.server.controllers.repository:ERROR: (26566-93472)   File "/home/vagrant/devel/pulp/server/pulp/plugins/util/publish_step.py", line 293, in _process_block
pulp.server.controllers.repository:ERROR: (26566-93472)     self.process_main()
pulp.server.controllers.repository:ERROR: (26566-93472)   File "/home/vagrant/devel/pulp_rpm/plugins/pulp_rpm/plugins/distributors/yum/publish.py", line 1023, in process_main
pulp.server.controllers.repository:ERROR: (26566-93472)     raise PulpCodedException(message=result_string)
pulp.server.controllers.repository:ERROR: (26566-93472) PulpCodedException: A general pulp exception occurred
pulp.server.async.tasks:INFO: Task failed : [78e433bd-5018-4ab4-9fc0-d318e52e110c] : A general pulp exception occurred
```